packages/uikit/src/pages/browser/CategoryBlock.tsx

```diff
@@ -0,0 +1,142 @@
+        transform: translateX(2px);
+    }
+`;
+
+export const CategoryBlock: FC<{ category: PromotionCategory; className?: string }> = ({
+    category,
+    className
+}) => {
+    const [containerRef, { width: w }] = useElementSize();
+    const width = w - 36;
+    const groups = useMemo(
+        () =>
+            category.apps.reduce((acc, app, index) => {
+                if (index % 3 === 0) {
+                    acc.push([app]);
+                } else {
+                    acc[acc.length - 1].push(app);
+                }
+                return acc;
+            }, [] as PromotedApp[][]),
+        [category.apps]
+    );
+
+    const groupsKeys = useMemo(() => groups.map(group => group.map(i => i.url).join('')), [groups]);
+    const canExpand = groups.length > 1;
+
+    return (
+        <div className={className} ref={containerRef}>
+            <Heading>
+                <H3>{category.title}</H3>
+                {canExpand && (
+                    <Link to={'.' + BrowserRoute.category + '/' + category.id}>
+                        <AllButton>
+                            <Label1>All</Label1>
+                        </AllButton>
+                    </Link>
+                )}
+            </Heading>
+            {canExpand ? (
+                <Carousel gap="8px" infinite={false}>
+                    {groups.map((group, groupIndex) => (
+                        <ListBlockStyled
+                            key={groupsKeys[groupIndex]}
+                            width={
+                                groupIndex === 0 || groupIndex === groups.length - 1
+                                    ? (width - 28).toString() + 'px'
+                                    : 'unset'
+                            }
+                            marginLeft={groupIndex === 0 ? '-34px' : '0'}
+                        >
+                            {group.map(item => (
+                                <CategoryGroupItem key={item.url} item={item} />
+                            ))}
+                        </ListBlockStyled>
+                    ))}
+                </Carousel>
+            ) : (
+                groups.map((group, groupIndex) => (
+                    <ListContainer key={groupsKeys[groupIndex]}>
+                        <ListBlockStyled key={groupsKeys[groupIndex]} width="100%">
+                            {group.map(item => (
+                                <CategoryGroupItem key={item.url} item={item} />
+                            ))}
+                        </ListBlockStyled>
+                    </ListContainer>
+                ))
+            )}
+        </div>
+    );
+};
+
+export const CategoryGroupItem: FC<{ item: PromotedApp }> = ({ item }) => {
+    const ref = useOpenLinkOnAreaClick(item.url);
+
+    return (
+        <ListItemStyled key={item.url} ref={ref}>
+            <PromotedItem>
+                <PromotedItemImage src={item.icon} />
+                <PromotedItemText>
```

Comment:
<img width="641" alt="Screenshot 2024-01-22 at 3 33 34 PM" src="https://github.com/tonkeeper/tonkeeper-web/assets/22742808/b62521f4-5333-4b42-83ab-1a060e0617df">

Could you please update layout to make items same by height?
description text should be max 2 lines ellipsis